hudi-client/src/main/java/org/apache/hudi/metrics/JmxMetricsReporter.java

```diff
@@ -82,6 +102,106 @@ public void report() {
 
   @Override
   public Closeable getReporter() {
-    return null;
+    return jmxServer.getReporter();
+  }
+
+  @Override
+  public void stop() {
+    if (jmxServer != null) {
+      try {
+        jmxServer.stop();
+      } catch (IOException e) {
+        LOG.error("Failed to stop JMX server.", e);
+      }
+    }
+  }
+
+  /**
+   * JMX Server implementation that JMX clients can connect to.
+   *
+   * <p>Heavily based on j256 simplejmx project
```

Comment:
whats the licensing of this code?

Reply:
> whats the licensing of this code?

The ISC License (https://opensource.org/licenses/ISC)

Reply:
we would add the license to LICENSE file? @vinothchandar Let's get this on land.

Reply:
apache and MIT are what we can clear easily. others need more scrutiny. Can we re-implement this ourselves please?